Userland/Libraries/LibTLS/HandshakeClient.cpp

```diff
@@ -224,6 +226,43 @@ void TLSv12::build_rsa_pre_master_secret(PacketBuilder& builder)
     builder.append(outbuf);
 }
 
+void TLSv12::build_dhe_rsa_pre_master_secret(PacketBuilder& builder)
+{
+    auto dh = m_context.server_diffie_hellman_params;
+    auto dh_p = Crypto::UnsignedBigInteger::import_data(dh.p.data(), dh.p.size());
+    auto dh_g = Crypto::UnsignedBigInteger::import_data(dh.g.data(), dh.g.size());
+    auto dh_Ys = Crypto::UnsignedBigInteger::import_data(dh.Ys.data(), dh.Ys.size());
+    auto dh_key_size = dh.p.size();
+
+    auto dh_random = Crypto::NumberTheory::random_number(0, dh_p);
+    auto dh_Yc = Crypto::NumberTheory::ModularPower(dh_g, dh_random, dh_p);
```

Comment:
I know this might sound like a roundabout way to do a simple modexp, but could we move this to `Crypto::Exchange::DiffieHellman` along with the key structure? (probably something similar to https://github.com/alimpfard/serenity/commit/9c7177ebd13400099dd5c35bafb77f43cbd27312#diff-c8328a7c38c75730469acaadeace67d28684b1552e3c6dddfb9f6e689b20984bR42)

my thinking here is that this could allow other applications to use DH without "implementing" it every time, and we could also add tests for it :P 

We could also address this in a future PR, though I'd prefer having tests for it here

Reply:
I like this idea. Let me see what I can accomplish.

Reply:
Let's call this PR good, we can mess with a refactoring to a DH class later.